src/protocol/sort/apply_sort/mod.rs

```diff
@@ -3,66 +3,51 @@ pub mod shuffle;
 use crate::{
     error::Error,
     ff::Field,
-    protocol::{
-        context::{Context, SemiHonestContext},
-        sort::{apply::apply_inv, generate_permutation::shuffle_and_reveal_permutation},
-    },
-    secret_sharing::Replicated,
+    protocol::{context::Context, sort::apply::apply_inv},
+    secret_sharing::SecretSharing,
 };
 
 use crate::protocol::sort::ApplyInvStep::ShuffleInputs;
-use crate::protocol::sort::SortStep::ShuffleRevealPermutation;
 
 use self::shuffle::{shuffle_shares, Resharable};
 
-#[derive(Debug)]
-pub struct SortPermutation<F: Field>(pub Vec<Replicated<F>>);
+use super::generate_permutation::RevealedAndRandomPermutations;
 
-impl<F: Field> SortPermutation<F> {
-    #[allow(dead_code)]
-    /// ## Panics
-    /// It will propagate panics from sort
-    /// # Errors
-    /// it will propagate errors from sort
-    pub async fn apply<I>(
-        self,
-        ctx: SemiHonestContext<'_, F>,
-        input: Vec<I>,
-    ) -> Result<Vec<I>, Error>
-    where
-        I: Resharable<F, Share = Replicated<F>>,
-    {
-        let revealed_and_random_permutation = shuffle_and_reveal_permutation(
-            ctx.narrow(&ShuffleRevealPermutation),
-            input.len().try_into().unwrap(),
-            self.0,
-        )
-        .await?;
+/// # Errors
+/// Propagates errors from shuffle/reshare
+pub async fn apply<C, F, S, I>(
```

Comment:
`apply` is a fairly generic name, and it is the same as the one in `apply.rs`. Can we give this a unique name, like `apply_permutation` or something?

Reply:
Just making it apply_sort_permutation for now. Earlier we were calling it through the object SortPermutation but now that wrapper is gone, this should make more sense (I think)